cas-server-core/src/main/java/org/jasig/cas/ticket/TicketGrantingTicketImpl.java

```diff
@@ -151,6 +152,10 @@ public synchronized ServiceTicket grantServiceTicket(final String id,
         final List<Authentication> authentications = getChainedAuthentications();
         service.setPrincipal(authentications.get(authentications.size()-1).getPrincipal());
 
+        if (registeredService.isOnlyKeepLatestST()) {
+            // remove this service if it already exists to only keep the latest service ticket for a service
+            this.services.values().remove(service);
```

Comment:
This is where the hostname/path comparison comes into play to compare services based on ids.